packages/react-native-web/src/exports/Modal/ModalPortal.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import { useEffect, useMemo } from 'react';
+
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+
+import ReactDOM from 'react-dom';
+
+export type ModalPortalProps = {|
+  children: any
+|};
+
+function ModalPortal(props: ModalPortalProps) {
+  const { children } = props;
+
+  // Only create the element once.
+  const element = useMemo(() => {
+    if (canUseDOM) {
+      return document.createElement('div')
+    }
+  }, []);
+
+  useEffect(() => {
+    if (canUseDOM && element && document.body) {
+      document.body.appendChild(element);
+    }
+
+    return () => {
+      if (canUseDOM && element && document.body) {
+        document.body.removeChild(element);
+      }
+    }
+  }, [element]);
+
+  if (!canUseDOM || !element) {
```

Comment:
Depending on the result of useCanDom, there was a difference between the server side and the client side DOM to be rendered. So I solved this problem by creating the element in useEffect and modifying the first rendering result for the server side and the client side in the same way.

```suggestion
  const [element, setElement] = useState()

  useEffect(() => {
    const newElement = document.createElement('div')
    setElement(newElement)
    document.body.appendChild(newElement);
    return () => {
      document.body.removeChild(newElement);
    }
  }, [])

  if (!element) {
```

Reply:
Perhaps I don't understand how `next.js` works - won't this fail because it can't use the DOM?

Reply:
Ah, found this - https://github.com/vercel/next.js/tree/canary/examples/with-portals reading through it now

Reply:
SSR does not execute useEffect, and only when JS is executed in the browser will it be processed in useEffect. So it won't fail.

Reply:
Ok - I've implemented per that - I had to add some extra checks, unfortunately.  The flow-typing failed if I didn't guard against some scenarios.

Reply:
I can confirm that it works well on SSR! Thanks! 🎉